.github/workflows/superset-python.yml

```diff
@@ -0,0 +1,174 @@
+name: Python
+
+on:
+  # only build on direct push to `master` branch
+  push:
+    branches: [ master ]
+    paths:
+    - ./**/*.py
+    - superset/**
+    - tests/**
+    - requirements*.txt
+  # but also build on pull requests to any branch
+  # (the so-called feature branch)
+  pull_request:
+    paths:
+    - ./**/*.py
+    - superset/**
+    - tests/**
+    - setup.py
+    - requirements*.txt
+
+jobs:
+  lint:
+    runs-on: ubuntu-18.04
+    strategy:
+      matrix:
+        python-version: [3.6]
+    env:
+      PYTHON_LINT_TARGET: setup.py superset tests
+      CI: github-actions
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+    - name: Setup Python
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      uses: apache-superset/cached-dependencies@ddf7d7f
+    - name: black
+      run: black --check $(echo $PYTHON_LINT_TARGET)
```

Comment:
@craig-rueda and @ktmud apologies for chiming in late here but should we be invoking `tox` here rather than redefining the various CI commands?

Reply:
I didn't touch tox because I didn't want to interfere with Travis. Each tox env seems to be installing its own requirements, etc, which makes it harder to split steps and reuse caches like we did in the current GH Actions setup.

Reply:
Most of these commands are one liners so I wouldn't worry too much, but feel free to refactor if you could find a way to reuse `tox` commands without degrading the building speed.